src/frontend/classic/components/Viewer.svelte

```diff
@@ -40,11 +44,81 @@
     function toggleFullScreen() {
         if (!document.fullscreenElement) {
             document.documentElement.requestFullscreen();
-        } else {
-            if (document.exitFullscreen) {
-                document.exitFullscreen();
+        } else if (document.exitFullscreen) {
+            document.exitFullscreen();
+        }
+    }
+
+    function loadThumbnail(el: Element) {
+        const div = el as HTMLDivElement;
+        const src = div.getAttribute("data-image");
+
+        if (src) {
+            div.style.backgroundImage = `url('${src}')`;
+        }
+    }
+
+    function loadWide(el: Element) {
+        const img = el as HTMLImageElement;
+        const src = img.getAttribute("data-image");
+
+        if (src) {
+            img.src = src;
+        }
+    }
+
+    function loadImages(currentIndex: number): void {
```

Comment:
Should use a simple foreach with await instead of a recursive function for a sequential processing.